src/Chapter3.hs

```diff
@@ -970,6 +1039,33 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Weekday = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving (Eq, Show)
+
+instance Enum Weekday where
+  succ Monday    = Tuesday
+  succ Tuesday   = Wednesday
+  succ Wednesday = Thursday
+  succ Thursday  = Friday
+  succ Friday    = Saturday
+  succ Saturday  = Sunday
+  succ Sunday    = Monday
+
+daysToParty :: Weekday -> Int
+daysToParty Monday    = 4
+daysToParty Tuesday   = 3
+daysToParty Wednesday = 2
+daysToParty Thursday  = 1
+daysToParty Friday    = 7
+daysToParty Saturday  = 6
+daysToParty Sunday    = 5
```

Comment:
This could be solved with `Enum` instance, if you write the instance properly with all necessary methods or derive instance 🙂